keymanapp/keyman
# fix(web): Allow touch-alias elements to clear text selection

Fixes #5853 

Original issue was reported on Keyman for Android (hence the android scope on the branch name), but the underlying behavior seems to be from KeymanWeb engine.

When the user selects text and presses a key, the expectation is for the selected text to be replaced by the processed keystroke.
Currently when selected text is passed to KeymanWeb, the output includes both selected text and keystroke. Effectively, the "selected text" remains.

If the user selects text beyond the working text area, I don't think we want to do any deletion/replacement.

## User Testing
For this test, load the web/testing/prediction-mtnt/ test page (includes English suggestion banner with English keyboard). Use Chrome browser in mobile viewport with the Developer console open (F12)
Ignore console exceptions about `Uncaught TypeError` since they're resolved by #5873

* **TEST_WEB_MOBILE** - Test text selection and replacement
1. In the default text area, type a few sentences
2. Click on the test page outside of all the text areas/ text fields
3. Hold down <kbd>shift</kbd> and click the text area to select enough text so the entire text selection includes both test page text and text from the text area
4. Type a key
5. Verify the selected text is **not** replaced
6. Verify the typed key is inserted into the text area
7. Click in the original text area
7. Hold down <kbd>shift</kbd> and click the text area to select enough text so the entire text selection is in the text area
8. Type a key
9. Verify the selected text is replaced with the typed key

* **TEST_WEB_SUGGESTION** - Test text selection and suggestions
1. In the default text area, type a few sentences
2. Click on the test page outside of all the text areas/text fields
3. Hold down <kbd>shift</kbd> and click the text area to select enough text so the entire text selection includes both test page text and text from the text area
4. Click on a suggestion
5. Verify the word at the cursor is replaced with the suggested word
6. Click in the original text area
7. Hold down <kbd>shift</kbd> and click the text area to select enough text so the entire text selection is in the text area
8. Click on a suggestion
9. Verify the selected text is replaced with the suggestion
